db/compaction/compaction_service_test.cc

```diff
@@ -119,59 +252,70 @@ class CompactionServiceTest : public DBTestBase {
       }
     }
   }
+
+ private:
+  std::shared_ptr<Statistics> compactor_statistics_;
+  std::shared_ptr<Statistics> primary_statistics_;
+  std::shared_ptr<T> compaction_service_;
 };
 
-TEST_F(CompactionServiceTest, BasicCompactions) {
-  Options options = CurrentOptions();
-  options.env = env_;
-  options.statistics = CreateDBStatistics();
-  std::shared_ptr<Statistics> compactor_statistics = CreateDBStatistics();
-  options.compaction_service = std::make_shared<MyTestCompactionService>(
-      dbname_, options, compactor_statistics);
+typedef testing::Types<MyTestCompactionService, MyTestCompactionServiceLegacy>
+    MyTestCompactionServiceTypes;
+TYPED_TEST_CASE(CompactionServiceTest, MyTestCompactionServiceTypes);
```

Comment:
I feel like `TEST_P` is more understandable, even considering it'd need some redundancy in specifying the classes to test (maybe an enum for specifying the classes in `INSTANTIATE_TEST_SUITE_P` and then select the class to test accordingly in `CompactionServiceTest`). But this is fine too since it minimizes change to run on a new `CompactionService` subclass...

Reply:
Make sense, another benefit is we could remove all `CompactionServiceTest<TypeParam>` with TEST_P.